src/test/java/org/opentripplanner/middleware/utils/ItineraryUtilsTest.java

```diff
@@ -0,0 +1,160 @@
+        String queryWithRealtimeParam = BASE_QUERY + "&" + IGNORE_REALTIME_UPDATES_PARAM + "=false";
+        List<String> queries = List.of(BASE_QUERY, queryWithRealtimeParam);
+
+        for (String query : queries) {
+            String queryExcludingRealtime = ItineraryUtils.excludeRealtime(query);
+            Map<String, String> params = ItineraryUtils.getQueryParams(queryExcludingRealtime);
+            Assertions.assertEquals("true", params.get(IGNORE_REALTIME_UPDATES_PARAM));
+        }
+    }
+
+    @Test
+    public void testUpdateTripWithVerifiedItinerary() throws IOException, URISyntaxException {
+        String query = BASE_QUERY + "&ui_activeItinerary=1";
+
+        String mockResponse = FileUtils.getFileContents(
+            TEST_RESOURCE_PATH + "persistence/planResponse.json"
+        );
+        OtpDispatcherResponse otpDispatcherResponse = new OtpDispatcherResponse(mockResponse);
+        List<Itinerary> itineraries = otpDispatcherResponse.getResponse().plan.itineraries;
+
+        MonitoredTrip trip = new MonitoredTrip();
+        trip.id = "testUpdateTripWithVerifiedItinerary";
+        trip.queryParams = query;
+        // Write some itinerary that is not the one we want.
+        trip.itinerary = itineraries.get(0);
+
+        ItineraryUtils.updateTripWithVerifiedItinerary(trip, itineraries);
+
+        Assertions.assertEquals(itineraries.get(1), trip.itinerary);
+    }
+
+    private void testItineraryDepartsSameDay(String time, boolean expected, Long... startTimes) {
+        // Trip is in US Eastern timezone per the place set in makeBarebonesTrip().
+        MonitoredTrip trip = makeBarebonesTrip();
+        ZoneId zoneId = trip.tripZoneId();
+
+        // startTimes are in US Eastern timezone.
+        for (Long startTime : startTimes) {
+            Itinerary itinerary = new Itinerary();
+            Instant instant = Instant.ofEpochMilli(startTime);
+            itinerary.startTime = Date.from(instant);
+            Assertions.assertEquals(
+                expected,
+                ItineraryUtils.itineraryDepartsSameDay(itinerary, QUERY_DATE, time, zoneId),
+                String.format(
+                    "%s %s be considered same day as %s %s",
+                    ZonedDateTime.ofInstant(instant, zoneId),
+                    expected ? "should" : "should not",
+                    QUERY_DATE,
+                    time
+                )
+            );
+        }
+    }
+
+    @Test
+    public void testItineraryDepartsSameDay() {
+        // All times EDT (GMT-04:00)
+        testItineraryDepartsSameDay(QUERY_TIME, true,
+            1597302000000L, // August 13, 2020 3:00:00 AM
+            1597377599000L, // August 13, 2020 11:59:59 PM
+            1597388399000L // August 14, 2020 02:59:59 AM, considered to be Aug 13.
+        );
+        testItineraryDepartsSameDay("1:23", true,
+            1597215600000L, // August 12, 2020 3:00:00 AM
+            1597291199000L, // August 12, 2020 11:59:59 PM
```

Comment:
I'd actually expect both of these to return false. The logic is that the request date is still August 13th, so one would not expect an itinerary with a start time on August 12th to show up at all. Despite August 12th being the same service day at the time, it is not the same request date.

Reply:
Actually, the second one might make sense for a trip that was "arriving by" 1:23am.